File: front/src/types/widget.ts


export type Widget = {
    id: string;
    type: WidgetType;
    mount_point: MountPoint;
    name: string;
    config: any
}

export type WidgetType = "line-chart"

export type MountPoint = "root_widget" | "device_widget" | "capability_widget"

File: shared/plugin/widget.go
package plugin

type Widget struct {
	ID         string           `json:"id"`
	Type       WidgetType       `json:"type"`
	Name       string           `json:"name"`
	Config     map[string]any   `json:"config"`
	MountPoint WidgetMountPoint `json:"mount_point"`
}

type WidgetType string

type WidgetMountPoint string

const (
	DashboardWidget  WidgetMountPoint = "root_widget"
	DeviceWidget     WidgetMountPoint = "device_widget"
	CapabilityWidget WidgetMountPoint = "capability_widget"
)

const (
	TypeLineChart WidgetType = "line-chart"
	TypeKVList    WidgetType = "key-value-list"
)
